Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp

```diff
@@ -211,6 +211,20 @@ void ProcessLauncher::launchProcess()
     GRefPtr<GSubprocessLauncher> launcher = adoptGRef(g_subprocess_launcher_new(G_SUBPROCESS_FLAGS_INHERIT_FDS));
     g_subprocess_launcher_take_fd(launcher.get(), socketPair.client, socketPair.client);
 
+#if USE(SYSPROF_CAPTURE)
+    int sysprofFd = -1;
+
+    if (const char* sysprofFdStr = getenv("SYSPROF_CONTROL_FD"))
+        sysprofFd = atoi(sysprofFdStr);
```

Comment:
We have a `parseInteger()` template in WTF that returns an optional, and that allows checking whether the input was valid, so I would do instead:

```suggestion
        sysprofFd = parseInteger<int>(sysprofFdStr).value_or(-1);
```

There's also `String::number(int)` to convert a number to a string but probably not worth using below as it would need the usual `.utf8().data()` dance to be passed to GLib functions.